src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -505,14 +553,27 @@
                       public-key
                       ca-exts)
         cacrl       (utils/generate-crl (.getIssuerX500Principal cacert)
+                                        private-key
+                                        public-key)
+        infra-crl   (utils/generate-crl (.getIssuerX500Principal cacert)
                                         private-key
                                         public-key)]
     (write-cert-to-inventory! cacert (:cert-inventory ca-settings))
     (utils/key->pem! public-key (:capub ca-settings))
     (utils/key->pem! private-key
       (create-file-with-perms (:cakey ca-settings) private-key-perms))
     (utils/cert->pem! cacert (:cacert ca-settings))
-    (utils/crl->pem! cacrl (:cacrl ca-settings))))
+    (utils/crl->pem! cacrl (:cacrl ca-settings))
+    (utils/crl->pem! infra-crl (:infra-crl ca-settings)))
+
+    ;; Generate a file containing serial numbers of the configured compiler master certificates, if any.
+    ;; Open - Would this be a right place to generate this as we want the compile master serials list to
+    ;; get generated everytime the compile masters are updated. And looking at this function this appears
+    ;; more like a one-time only method (generates ca keys, cert.
+    ;; If so, what would be an appropriate place for doing the below which can get called as many times as
+    ;; needed.
+    (generate-compile-master-serials (:compile-masters ca-settings) (:compile-master-serials ca-settings))
```

Comment:
you may want to look into registering a filesystem watcher on the compile-masters file and running this when it changes.

Reply:
See the certificate-authority-service for an example of doing so with the crl.

Reply:
@justinstoller 
Am looking into adding a file system watcher for infra nodes file and was not clear on some things.  Questions on code https://github.com/puppetlabs/puppetserver/blob/master/src/clj/puppetlabs/services/ca/certificate_authority_service.clj#L46-56
- about the (when (not= ca-crl-file host-crl-file...) check. This seems to copy content of cacrl, anytime it changes, to local/host crl file. Am guessing I don't need to maintain two files like this. 
- the code seems to register a watcher for "changed-path" (?) type of event on ca-crl-file or may be it is saying "ca-crl-file" was the changed file within the watched directory (its parent). I am interested in "file got updated" type of event for infra-nodes. Do I also need to watch the directory housing infra-nodes file and check if it was the infra-nodes file that changed then invoke the callback?